### PR TITLE
Add fraction support to lexer.

### PR DESCRIPTION
**Pull request by wmedrano**
Tested with lexer unit tests and updating `math.scm` to use the new syntax.

### CODE REVIEW

**Comment by wmedrano** [comment]
So, not ready to merge yet. Added some unit tests  that show weird behavior. Will report back when resolved.

File: crates/steel-core/src/primitives/nums.rs [hunk 1]
```diff
@@ -404,15 +406,6 @@ impl NumOperations {
     }
 }
 
-impl IntoSteelVal for num::BigInt {
```
Comment (wmedrano) [inline crates/steel-core/src/primitives/nums.rs:407]: This was moved to be next to the other `IntoSteelVal` for basic types.

File: crates/steel-core/src/tests/mod.rs [hunk 2]
```diff
@@ -55,7 +55,6 @@ test_harness_success! {
     babbage_problem,
     balanced_brackets,
     basic_apply,
-    bignum,
```
Comment (wmedrano) [inline crates/steel-core/src/tests/mod.rs:58]: Merged into math.scm test.

**Comment by wmedrano** [comment]
If tests are good, then i think this is ready

File: crates/steel-core/src/primitives/nums.rs [hunk 3]
```diff
@@ -84,10 +84,10 @@ fn multiply_2_impl(x: &SteelVal, y: &SteelVal) -> Result<SteelVal> {
             res *= y.as_ref();
             res.into_steelval()
         }
-        (SteelVal::BigFract(x), SteelVal::BigFract(y)) => (x.as_ref() + y.as_ref()).into_steelval(),
+        (SteelVal::BigFract(x), SteelVal::BigFract(y)) => (x.as_ref() * y.as_ref()).into_steelval(),
```
Comment (wmedrano) [inline crates/steel-core/src/primitives/nums.rs:87]: Big oops :facepalm:

Comment (mattwparas) [reply crates/steel-core/src/primitives/nums.rs:87]: I missed it in the last one, eyes aren't good enough to see it :shrug:

**Comment by mattwparas** [comment]
lgtm pending the tests

**Review by mattwparas — APPROVED** [review]

**Pull request merged**